S7.Net/PLC.cs

```diff
@@ -107,11 +107,35 @@ public bool IsConnected
             {
                 try
                 {
+                    bool clientConnected = false;
+
                     if (tcpClient == null)
                         return false;
 
                     //TODO: Actually check communication by sending an empty TPDU
-                    return tcpClient.Connected;
+                    if (tcpClient.Client.Poll(0,SelectMode.SelectRead))
+                    {
+                        byte[] buffer = new byte[1];                        
+                        if (tcpClient.Client.Receive(buffer, SocketFlags.Peek) == 0)
+                        {
+                            // client disconnected
+                            clientConnected = false;                            
```

Comment:
you can just return false here, and skip the else branch. That also removes the need for the clientConnected variable.